### PR TITLE
Cppcheck 9.1

### PR DESCRIPTION
**Pull request by drwells**
Running cppcheck with inconclusive generates a lot of noise but also some more useful stuff.

### CODE REVIEW

File: include/deal.II/lac/sparsity_pattern.h [hunk 1]
```diff
@@ -884,22 +884,6 @@ class SparsityPattern : public SparsityPatternBase
...
-   * sparsity pattern using the add() member function, and are removed when
-   * calling compress().
-   *
-   * You should not assume that the variable declared here has a certain
-   * value. The initialization is given here only to enable the compiler to
-   * perform some optimizations, but the actual value of the variable may
-   * change over time.
-   */
-  static const size_type invalid_entry = SparsityPatternBase::invalid_entry;
-
```
Comment (kronbichler) [inline include/deal.II/lac/sparsity_pattern.h]: This causes an error on the MacOSX builder:
```
source/lac/sparsity_pattern.cc:37:53: error: no member named 'invalid_entry' in 'dealii::SparsityPattern'

  const SparsityPattern::size_type SparsityPattern::invalid_entry;
```

**Comment by drwells** [comment]
I am fixing this now.

**Comment by drwells** [comment]
OK, that should do it. Lets see if the tester is happy.

**Comment by masterleinad** [comment]
The indentation is off.

**Comment by drwells** [comment]
Fixed (thanks!)

**Pull request merged**